assets/src/components/shuttleMapPage.tsx

```diff
@@ -57,39 +74,160 @@ const ShuttleMapPage = (): ReactElement<HTMLDivElement> => {
+
+  const followPositions = useMemo(
+    () =>
+      followerController
+        ? {
+            "user-location": location
+              ? [latLng(location.latitude, location.longitude)]
+              : [],
+            "vehicle-location": positions,
+          }[followerController]
+        : [],
+    [followerController, location, positions]
+  )
+
+  useEffect(() => {
+    if (followerController === false) {
+      setShouldFollow("vehicle-location")
+    }
+    // This effect is an "event", and not "reactive" to all it's dependencies.
+    // Because we _only_ want to run this when the run id's change, we need to
+    // suppress the exhaustive-deps lint until `useEffectEvent` is stabilized
+    //
+    // See the react docs:
+    // > ## Is it okay to suppress the dependency linter instead?
+    // > After `useEffectEvent` becomes a stable part of React,
+    // > we recommend never suppressing the linter.
+    // > -- https://react.dev/learn/separating-events-from-effects#is-it-okay-to-suppress-the-dependency-linter-instead
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedShuttleRunIds])
+
+  const followerFn = usePickerContainerFollowerFn()
+
+  return (
+    <div className="c-shuttle-map__map">
+      <Map
+        stateClasses={FollowerStatusClasses(
+          followerController == "vehicle-location"
+        )}
+        vehicles={selectedShuttles}
+        selectedVehicleId={selectedVehicleId}
+        shapes={selectedRouteShapes}
+        trainVehicles={trainVehicles}
+        onPrimaryVehicleSelect={selectVehicle}
+        tileType={tileType}
+      >
+        <LayersControl.WithTileContext setTileType={setTileType} />
+
+        <UserLocationControl
+          title="Show your current location"
+          onClick={() => {
+            setShouldFollow("user-location")
+          }}
+          disabled={followerController == "user-location"}
+        />
+        <RecenterControl
+          position="topright"
+          recenter={() => setShouldFollow("vehicle-location")}
+        />
+
+        <>
+          {(followerController === "user-location" || location) && (
+            <UserLocation onLocationUpdate={(l) => setLocation(l)} />
+          )}
+        </>
+        <>{location && <UserLocationMarker location={location} />}</>
+
+        <InterruptibleFollower
+          {...state}
+          setShouldFollow={(shouldFollow) => {
+            shouldFollow === false && setShouldFollow(false)
+          }}
+          positions={followPositions}
+          onUpdate={(...args) => {
+            followerFn(...args)
+          }}
+        />
+      </Map>
     </div>
   )
 }
 
+interface UserLocationProps {
+  onLocationUpdate: Dispatch<GeolocationCoordinates | null>
+}
+const UserLocation = ({ onLocationUpdate }: UserLocationProps) => {
+  const location = useGeolocation()
+  useEffect(() => {
+    onLocationUpdate(location)
+  }, [onLocationUpdate, location])
+  return null
```

Comment:
@mbta/skate-developers this is the logic I think is somewhat weird. I needed a way to conditionally call this hook, because we don't want to ask for permission for the user's location until they press the location button. 

I think it's fine, but some extra scrutiny here might be warranted. Though, I do think I prefer this solution instead of say, adding a parameter to `useGeolocation` to conditionally call `watchPosition`.

Reply:
There is a test that explicitly tests that `watchPosition` is not called until the button is clicked though.

Reply:
I think this is fine. Since hooks can't be called conditionally our options are to put the conditional inside of the hook based on an argument, or to conditionally render / not render the component calling the hook, and I think either option is all right depending on the context.